### PR TITLE
GEODE-4712 GEODE-5943: shut down the bucketSorter when destroying the partitioned region

### PR DESCRIPTION
**Pull request by jinmeiliao**
Thank you for submitting a contribution to Apache Geode.

In order to streamline the review of the contribution we ask you
to ensure the following steps have been taken:

### For all changes:
- [ ] Is there a JIRA ticket associated with this PR? Is it referenced in the commit message?

- [ ] Has your PR been rebased against the latest commit within the target branch (typically `develop`)?

- [ ] Is your initial contribution a single, squashed commit?

- [ ] Does `gradlew build` run cleanly?

- [ ] Have you written or updated unit tests to verify your changes?

- [ ] If adding new dependencies to the code, are these dependencies licensed in a way that is compatible for inclusion under [ASF 2.0](http://www.apache.org/legal/resolved.html#category-a)?

### Note:
Please ensure that once the PR is submitted, you check travis-ci for build issues and
submit an update to your PR as soon as possible. If you need help, please send an
email to dev@geode.apache.org.

### CODE REVIEW

**Comment by jinmeiliao** [comment]
@kirklund I talked to Eric about this. He thinks bouncing the VM should be the approach here. But I think there may be other ways as well. maybe we can pair on this to find a better approach. I know why this happened, but I am not sure what's the correct way to fix the flakiness.

**Comment by kirklund** [comment]
Yesterday, Jinmei and I discovered that the JVM which goes critical and causes the failure has 12 GemFireCacheImpl instances in it. So this test seems to be hitting some sort of memory leak that needs to be fixed (not sure if it's a test or product leak).

File: geode-core/src/main/java/org/apache/geode/internal/cache/PartitionedRegion.java [hunk 1]
```diff
@@ -7633,6 +7633,8 @@ protected void postDestroyRegion(boolean destroyDiskRegion, RegionEventImpl even
       colocatedWithRegion.getColocatedByList().remove(this);
     }
 
+    bucketSorter.shutdown();
```
Comment (agingade) [inline geode-core/src/main/java/org/apache/geode/internal/cache/PartitionedRegion.java]: Needs null check. Its created if the eviction is configured.

File: geode-core/src/main/java/org/apache/geode/internal/cache/PartitionedRegion.java [hunk 2]
```diff
@@ -803,7 +803,7 @@ public PartitionedRegion(String regionName, RegionAttributes regionAttributes,
     if (this.getEvictionAttributes() != null
         && this.getEvictionAttributes().getAlgorithm().isLRUHeap()) {
       this.sortedBuckets = new ArrayList<BucketRegion>();
-      this.bucketSorter = LoggingExecutors.newScheduledThreadPool("BucketSorterThread", 1);
+      this.bucketSorter = LoggingExecutors.newScheduledThreadPool("BucketSorterRunnable", 1);
```
Comment (dschneider-pivotal) [inline geode-core/src/main/java/org/apache/geode/internal/cache/PartitionedRegion.java]: This one should not be "BucketSorterRunnable". This is the thread name so the original "BucketSorterThread" was fine.

Comment (dschneider-pivotal) [reply geode-core/src/main/java/org/apache/geode/internal/cache/PartitionedRegion.java]: I unit test would be great but not required.

Comment (jinmeiliao) [reply geode-core/src/main/java/org/apache/geode/internal/cache/PartitionedRegion.java]: added the unit test and renamed the thread

**Pull request merged**